test/rebalancer-test.js

```diff
@@ -0,0 +1,267 @@
+const { expect, assert } = require("chai");
+const { ethers } = require("hardhat");
+
+let dai, aDai, usdc, awMatic;
+let aavePool, aaveAdapter, aaveProvider;
+let testAccount, testSigner;
+
+let DAI_CONTRACT = "0x8f3cf7ad23cd3cadbd9735aff958023239c6a063";
+let ADAI_CONTRACT = "0x27F8D03b3a2196956ED754baDc28D73be8830A6e";
+let wMATIC_CONTRACT = "0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270";
+let awMATIC_CONTRACT = "0x8dF3aad3a84da6b69A4DA8aeC3eA40d9091B2Ac4";
+let LENDING_POOL = "0x8dff5e27ea6b7ac08ebfdf9eb090f32ee9a30fcf";
+let USDC_CONTRACT = "0x2791bca1f2de4661ed88a30c99a7a9449aa84174";
+
+describe("AaveAdapter", function () {
+  before(async () => {
+    // account used for all the tests
+    [testSigner] = await ethers.getSigners();
+    testAccount = testSigner.address;
+
+    // get contracts
+    dai = await ethers.getContractAt("IERC20", DAI_CONTRACT, testSigner);
+    aDai = await ethers.getContractAt("IERC20", ADAI_CONTRACT, testSigner);
+    usdc = await ethers.getContractAt("IERC20", USDC_CONTRACT, testSigner);
+    awMatic = await ethers.getContractAt("IERC20", awMATIC_CONTRACT, testSigner);
+    aavePool = await ethers.getContractAt("ILendingPool", LENDING_POOL, testSigner);
+    aaveProvider = await ethers.getContractAt(
+      "IProtocolDataProvider",
+      "0x7551b5D2763519d4e37e8B81929D336De671d46d",
+      testSigner
+    );
+
+    // deploy Aave adapter
+    const AaveAdapter = await ethers.getContractFactory("AaveAdapter");
+    aaveAdapter = await AaveAdapter.deploy();
+    await aaveAdapter.deployed();
```

Comment:
After deploying it let's create's another instance of this contract with our `IAdapter` interface artifact. This way we will be sure that the contract implements the interface as intended. If required we can make changes to function modifier in interface definition.

Reply:
Done